### DISCUSSION TITLE
Import Speed - too slow, issues on my end?

### DISCUSSION THREAD

**Discussion by gluk1421** [vzhd1701/enex2notion#56] [Q&A]
The upload speed ranges from 1 second per block to up to 30 seconds per block.  At 1 second per block, it would take me about a month to upload all of my notebooks (around 4,000 notes).  I have read elsewhere that others have been able to upload thousands of notes in a matter of hours.

This seems unreasonably slow which makes me think something is wrong on my end.  My Internet speed is 100 mbps - so this does not seem to be the limiting factor.

I am located in Thailand IP address - is it possible Notion is limiting upload rates?  I can also switch to a VPN.

I started the upload on a free account but upgraded to the Personal Pro account in the middle of the first batch upload - which is still ongoing.  Is it possible Notion limits upload rates due to it being a free account?

Any other suggestions on how to speed this up?

How many seconds per block have others experienced?

=================

Edit: As another user suggested, the slow speed can work because you can run multiple powershell windows in parallel; I have started opening multiple powershells and there is no decrease in speed.  Theoretically I can open one for each notebook and the entire operation can be completed within 24 hours ...

**Comment by vzhd1701** [comment]
I am afraid that this is normal. The tool is slow, sorry about it.

It all depends on how many blocks your notes contain and what kind of blocks are those. Roughly speaking every paragraph in your Evernote note equals to one Notion block. Special elements like images and tables take longer than others to upload. So if you have simple notes with 1-2 paragraphs of plain text, then they will upload faster. If your notes are heavy with tables and images then you should expect longer upload time.

You are correct, the only thing you can do to speed it up is to run uploads in parallel, one upload per notebook. Buying Notion premium won't help here unfortunately.

**Reply by gluk1421** [reply]
Thanks, the parallel PowerShell instances worked great - setup one for each notebook and left in on overnight and it was done by morning.

Seems the notes that took the longest were the long articles with lots of text amounting to 1,000+ blocks.  The larger ones with image attachments were much shorter in block length even though they took up more storage space.  It took the same amount of time to upload 2 GB worth of largely image files as it took to upload 80 MB of text notes …

**Comment by pedrosimao** [comment]
I am trying to transfer all my webclips from Evernote, and `enex2notion` is incredibly slow, unhappily.
It's been 1h since I started importing and I am still on the 7th note. Maybe there is an extra anomaly on my side, besides the usual slow speed?
I know this are limitations of the Notion API, but I am trying to come-up with ideas to make a fast import for webclips specifically.
Maybe a good way would be to convert the Evernote webclips into Word `docx` instead of `PDF` and then use some kind of E2E library (Puppeteer) to import the `docx` file, which Notion converts automatically in a second. I have taken a look at the Notion API and I did not find a way to do the Word import (which is different than attaching a document).
